### ISSUE TITLE
Incorrect formatting of the DISTINCT ON construct

### ISSUE THREAD

**Issue by become-iron** [klahnakoski/mo-sql-parsing#41]
```python
>>> import mo_sql_parsing
>>> q = "SELECT DISTINCT ON (col) col, col2 FROM test"
>>> mo_sql_parsing.format(mo_sql_parsing.parse(q))
'SELECT DISTINCT ON(col) AS col, col2 FROM test'
```

`AS` is invalid part of the `DISTINCT ON` construct

Checked on PostgreSQL 13

**Comment by become-iron** [comment]
Well, should you really fix it if it's PostgreSQL's bug?

**Comment by klahnakoski** [comment]
It is not a postgres bug:  I got a snide remark, as I deserve. 

https://www.postgresql.org/message-id/CAKFQuwauxdgK%3D0gRiMUbMprcs83-NVgpZNvRkkNOKn_uF12Sjg%40mail.gmail.com

it turns out the geeks link was wrong.  I did some experimentation with postgres and I got a solution.  Here is the test

https://github.com/klahnakoski/mo-sql-parsing/commit/94b722cdf3034add76a38a4d02181b1f581951cf#diff-c27809480953c5908cdc63856a3e545a7fd67d739492c0968b61f5090a107b0cR113
